reactive/ovn_chassis_charm_handlers.py

```diff
@@ -133,7 +133,7 @@ def configure_nrpe():
 @reactive.when_none('charm.paused', 'is-update-status-hook')
 @reactive.when(OVN_CHASSIS_ENABLE_HANDLERS_FLAG,
                'ovsdb-subordinate.available',
-               'certificates.available')
```

Comment:
I'm a bit curious about this part. At present this reactive handler will run on every hook execution after the certificates appear, and the call to `publish_chassis_certificates` will be made each time.

We rely on juju behavior of not notifying our peer about the update unless the data actually changes.

So effectively this change would make us call `publish_chassis_certificates` more seldom, is that the intended change and is that really necessary? Or am I missing something?

Reply:
The difference in behavior comes from `charms.openstack` which installs a default handler for a set of states[0], for the case of `certificates.available` it will wire it to run the `configure_tls()`[1] method, so what I'm doing with this patch is to set the flag `ovn.certs.changed` in the body of the `configure_tls()` method to allow running `provide_chassis_certificates_to_principal()`

[0] https://github.com/openstack/charms.openstack/blob/master/charms_openstack/charm/defaults.py#L34-L51
[1] https://github.com/openstack/charms.openstack/blob/master/charms_openstack/charm/classes.py#L445